File: X-Clone-Backend/src/main/java/com/xclone/xclone/domain/post/Post.java
package com.xclone.xclone.domain.post;


import jakarta.persistence.*;

import java.sql.Timestamp;

@Entity
@Table(name = "posts")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "user_id")
    private Integer userId;

    @Column(name = "parent_id")
    private Integer parentId;

    @Column(length = 180)
    private String text;

    @Column(name = "created_at", updatable = false, insertable = false)
    private Timestamp createdAt;

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public Integer getId() {
        return id;
    }

    public Integer getUserId() {
        return userId;
    }

    public String getText() {
        return text;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Integer getParentId() {
        return parentId;
    }

    public void setParentId(Integer parentId) {
        this.parentId = parentId;
    }
}

File: X-Clone-Backend-FastAPI/app/models/post.py
from __future__ import annotations

from sqlalchemy import Column, Integer, String, TIMESTAMP

from ..db import Base


class Post(Base):
    __tablename__ = "posts"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, nullable=False)
    parent_id = Column(Integer, nullable=True)
    text = Column(String(180), nullable=True)
    created_at = Column(TIMESTAMP, nullable=True)


